program/tests/test_create_mint.rs

```diff
@@ -0,0 +1,450 @@
+use solana_program::program_error::ProgramError;
+use solana_program::program_pack::Pack;
+use solana_program::rent::Rent;
+use solana_program::system_program;
+use solana_sdk::account::Account;
+use solana_sdk::program_option::COption;
+use spl_token_2022::state::Mint;
+use spl_token_wrap::state::Backpointer;
+use spl_token_wrap::{
+    get_wrapped_mint_address, get_wrapped_mint_authority, get_wrapped_mint_backpointer_address,
+};
+use {
+    mollusk_svm::{result::Check, Mollusk},
+    solana_program::pubkey::Pubkey,
+    spl_token_wrap::instruction::create_mint,
+};
+
+const MINT_DECIMALS: u8 = 12;
+const MINT_SUPPLY: u64 = 500_000_000;
+const FREEZE_AUTHORITY: &str = "11111115q4EpJaTXAZWpCg3J2zppWGSZ46KXozzo9";
+
+fn setup_spl_mint(rent: &Rent) -> Account {
+    let state = spl_token::state::Mint {
+        decimals: MINT_DECIMALS,
+        is_initialized: true,
+        supply: MINT_SUPPLY,
+        freeze_authority: COption::Some(Pubkey::from_str_const(FREEZE_AUTHORITY)),
+        ..Default::default()
+    };
+    let mut data = vec![0u8; spl_token::state::Mint::LEN];
+    state.pack_into_slice(&mut data);
+
+    let lamports = rent.minimum_balance(data.len());
+
+    Account {
+        lamports,
+        data,
+        owner: spl_token::id(),
+        ..Default::default()
+    }
+}
+
+fn setup_token_2022_mint(rent: &Rent) -> Account {
+    let state = spl_token_2022::state::Mint {
+        decimals: MINT_DECIMALS,
+        is_initialized: true,
+        supply: MINT_SUPPLY,
+        freeze_authority: COption::Some(Pubkey::from_str_const(FREEZE_AUTHORITY)),
+        ..Default::default()
+    };
+    let mut data = vec![0u8; spl_token_2022::state::Mint::LEN];
+    state.pack_into_slice(&mut data);
+
+    let lamports = rent.minimum_balance(data.len());
+
+    Account {
+        lamports,
+        data,
+        owner: spl_token_2022::id(),
+        ..Default::default()
+    }
+}
```

Comment:
nit: spl_token and spl_token_2022's mints are exactly the same, so you can combine both of these functions by also taking the `owner` program id